listener/src/event_handler/flux_transition_handler.rs

```diff
@@ -89,8 +89,25 @@ impl EventHandler for FluxTransitionHandler {
                 ])
             },
             FluxTransitionType::Withdraw => {
-                warn!(logger, "Withdraw handling not yet implementd");
-                Ok(vec![])
+                let withdraw_query = util::entity_query(
+                    "Withdraw", EntityFilter::Equal("slot".to_string(), slot.to_value())
+                );
+                let withdraws = self.store
+                    .find(withdraw_query)?
+                    .into_iter()
+                    .map(PendingFlux::from)
+                    .collect::<Vec<PendingFlux>>();
+                account_state.apply_withdraws(&withdraws);
```

Comment:
Maybe we can return the account state from the match statement and have the logic to generate the entity operation vector shared between deposits and withdraws.

Reply:
Very good point! I will move this part out and learn how to return stuff from a match!